### DISCUSSION TITLE
Variablen & Schreibweise

### DISCUSSION THREAD

**Discussion by obermacker** [obermacker/spacefights#9] [General]
Ich bin da selbst inkonsequent, daher schließe ich mich Deiner Idee an, einheitliche Standards festzulegen:

Dies ist nur Brainstorming, nichts finalisiert.

Aus Deinem Commit:

> // for transitional period / compatiblity old vars
> $spieler_id = $playerID; $session_id = $sessionID; $username = $playerName;

entnehme ich:
```
$spieler_id := $playerID
$session_id := $sessionID
$username := $playerName
```

$ Erster Buchstabe klein, und bei zusammengesetzten Wörtern der jeweilige erste Buchstabe groß
Beispiel:` $playerMaxDamageInFight`

Wegwerf-Variablen oder Parameter bzw. Übergaben mit Bindestrich. Bsp. `$_disposableVariableAccordingToTheUsualRules`

Sprache: Englisch

Variablen: ausführlich; Regeln oben

Funktionen: Englisch ausführlich aber mit Bindestrichen (bsp. `function get_defense_count`)

Kommentare: Englisch (_Deutsche Kommentare nur im commit und vor dem merge übersetzen_.)

Wichtig ist, dass man Variablen nicht sinnlos abkürzt und man anhand der Bezeichnung erkennen kann, wofür sie da ist. Ausnahme sind altbekannte wie z.B. Exeption $e und `Zähler i`.

Gleiches gilt für den Code. 
Komplexe Algorithmen in eine Zeile gequetscht zu debuggen macht keine Freude.

Komplex heisst, es werden deutlich mehr Zeilen Code verwendet um ein Ergebnis zu erhalten als üblich oder die Parameterliste beinhaltet mehrere Übergaben oder mind. ein komplexes Array (z.B. aus Datenbankabfragen)

Ansonsten mag ich folgende Quotes von John Romero:

> Try to code transparently. Tell your lead and peers exactly how you are going to solve your current task and get feedback and advice. Do not treat game programming like each coder is a black box. The project could go off the rails and cause delays.

> Programming is a creative art form based in logic. Every programmer is different and will code differently. It's the output that matters.

Der letzte liegt mir am Herzen, es macht mir nämlich Freude Deinen Code anzuschauen.🍹

**Comment by schimmi70** [comment]
Ich bin auch für alles offen (wir müssen nur ein Regelwerk festschreiben) Bei der Bezeichnung von Variablen bin ich immer hin und her gerissen ob wie oben beschrieben oder mit Under Lines.... Allerdings finde ich das bei langen Namen blöd mehre Under Lines drinnen zu haben. Momentan nutze ich eher die Variante wie oben beschrieben. 
Wenn du mit dem oben genannten einverstanden bist, können wir das gerne zu festschreiben....

> Wichtig ist, dass man Variablen nicht sinnlos abkürzt und man anhand der Bezeichnung erkennen kann, wofür sie da ist.

des halb nutze ich gerade recht lange Variablen / Funktion Namen, ich denke das erlaubt es weiterhin den Code wenig zu dokumentieren.

> Komplexe Algorithmen in eine Zeile gequetscht zu debuggen macht keine Freude.

Ich neige oft dazu Zwischenschritte weg zu rationalisieren z.b. $result = mysqli(...)  / $row = fetch_object ($results)  in  $row = fetch_object (mysqli(...)) ich hoffe das stört dich nicht zu sehr !?
Ansonsten hasse ich schlecht formatierten Code, der schwer zu lesen ist ;)

> Komplex heisst, es werden deutlich mehr Zeilen Code verwendet um ein Ergebnis zu erhalten als üblich oder die Parameterliste beinhaltet mehrere Übergaben oder mind. ein komplexes Array (z.B. aus Datenbankabfragen)

Den Satz habe ich jetzt nicht wirklich verstanden, hast du da ein Beispiel aus dem Code ?
